src/webui/www/private/scripts/mocha-init.js

```diff
@@ -144,6 +144,19 @@ var initializeWindows = function() {
             height: loadWindowHeight(id, 420),
             onResize: function() {
                 saveWindowSize(id);
+            },
+            onClose: function() {
+                var hashString = location.hash && location.hash.replace(/^#/, '');
```

Comment:
From a static typing viewpoint (myself), it is a bit hard to understand what it really does.
I suppose it could be equivalent/rewritten to `var hashString = location.hash ? location.hash.replace(/^#/, '') : "";` ?
And please change other similar places too.